test/framework/toolchain.py

```diff
@@ -948,6 +948,32 @@ def test_nosuchtoolchain(self):
         tc = self.get_toolchain('intel', version='1970.01')
         self.assertErrorRegex(EasyBuildError, "No module found for toolchain", tc.prepare)
 
+    def test_mpi_exec_nranks(self):
+        """Test mpi_exec_nranks function."""
+        self.modtool.prepend_module_path(self.test_prefix)
+
+        tc = self.get_toolchain('gompi', version='2018a')
+        tc.prepare()
+        self.assertEqual(tc.mpi_exec_nranks(nr_ranks=2), "mpirun -n 2")
```

Comment:
Also check the default?

```python
self.assertEqual(tc.mpi_exec_nranks(), "mpirun -n 1")
```

And maybe make sure that we don't get a nasty traceback when a string value is passed:

```python
error_pattern = "pattern for error message goes here"
self.assertErrorRegex(EasyBuildError, error_pattern, tc.mpi_exec_nranks, '2')